src/udev/udev-builtin-net_id.c

```diff
@@ -794,6 +797,44 @@ static int names_mac(sd_device *dev, struct netnames *names) {
         return 0;
 }
 
+static int names_netdevsim(sd_device *dev, struct netnames *names) {
+        sd_device *netdevsimdev;
+        const char *sysname;
+        unsigned int addr;
+        const char *port_name = NULL;
+        size_t l;
+        char *s;
+        int r;
+
+        assert(dev);
+        assert(names);
+
+        r = sd_device_get_parent_with_subsystem_devtype(dev, "netdevsim", NULL, &netdevsimdev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_sysname(netdevsimdev, &sysname);
+        if (r < 0)
+                return r;
+
+        if (sscanf(sysname, "netdevsim%u", &addr) != 1)
+                return -EINVAL;
+
+        (void) sd_device_get_sysattr_value(dev, "phys_port_name", &port_name);
+
+        s = names->netdevsim_path;
+        l = sizeof(names->netdevsim_path);
+        l = strpcpyf(&s, l, "i%u", addr);
+        if (port_name)
+                l = strpcpyf(&s, l, "n%s", port_name);
+        else
+                return 0;
```

Comment:
Please don't use the `strpcpyf` functions, they are so hard to parse. What about
```C
    bool ok;
    if (port_name)
          ok = snprintf_ok(names->netdevsim_path, sizeof(names->netdevsim_path), "i%un%s", addr, port_name);
    else
          ok = snprintf_ok(names->netdevsim_path, sizeof(names->netdevsim_path), "i%u", addr);
    if (!ok)
          return -ENOBUFS;
    names->type = NET_NETDEVSIM;
    return 0;
```
This detects truncation properly and sets type always.

Reply:
I don't mind, I just copy&pasted and adjusted existing code. Will do it like you ask.